app/models/person.rb

```diff
@@ -241,15 +238,52 @@ def exported_key= new_key
     serialized_public_key = new_key
   end
 
-  #database calls
+  def username
+    @username ||= owner ? owner.username : diaspora_handle.split("@")[0]
+  end
+
+  def webfinger_hash
+    {
+      acct_uri:    "acct:#{diaspora_handle}",
+      alias_url:   "#{url}people/#{guid}",
+      hcard_url:   url[0...-1] + DiasporaFederation::Engine.routes.url_helpers.hcard_path(guid),
+      seed_url:    url,
+      profile_url: profile_url,
+      atom_url:    atom_url,
+      salmon_url:  receive_url,
+      guid:        guid,
+      pubkey:      serialized_public_key
+    }
+  end
+
+  def hcard_profile_hash
+    {
+      guid:             guid,
+      nickname:         username,
+      full_name:        "#{first_name} #{last_name}",
+      url:              url,
+      photo_large_url:  image_url,
+      photo_medium_url: profile.image_url_medium,
+      photo_small_url:  profile.image_url_small,
+      pubkey:           serialized_public_key,
+      searchable:       searchable,
+      first_name:       first_name,
+      last_name:        last_name
+    }
+  end
```

Comment:
I am not sure if I like those hashes. It would be nicer to just have a list of attributes that are included.